lib/main.coffee

```diff
@@ -0,0 +1,51 @@
+{CompositeDisposable} = require 'atom'
+
+module.exports =
+  config:
+    shellcheckExecutablePath:
+      type: 'string'
+      title: 'Shellcheck Executable Path'
+      default: 'shellcheck' # Let OS's $PATH handle the rest
+    enableNotice:
+      type: 'boolean'
+      title: 'Enable Notice Messages'
+      default: false
+
+  activate: ->
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe \
+     'linter-shellcheck.shellcheckExecutablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-shellcheck.enableNotice',
+      (enableNotice) =>
+        @enableNotice = enableNotice
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    helpers = require('atom-linter')
+    provider =
+      grammarScopes: ['source.shell']
+      scope: 'file'
+      lintOnFly: true
+      lint: (textEditor) =>
+        filePath =  textEditor.getPath()
+        showAll = @enableNotice
+        parameters = ['-f', 'gcc', filePath ]
+        return helpers.exec(@executablePath, parameters).then (output) ->
+          regex = /.+?:(\d+):(\d+):\s(\w+?):\s(.+)/g
```

Comment:
I am low on caffeine I guess, regex is all fine, but I don't get any errors with
```sh
SOME="thing"
```
however the cli reports this
```
/var/www/html/ex/.idea/test.sh:1:1: note: SOME appears unused. Verify it or export it. [SC2034]
```